[FIX] sale_timesheet: missing other_costs sequence in project update

sabr-odoo opened the pull request:
when the user tries to access the project update
View of a project and at least one timesheet of that project has timesheet_invoice_type='other_costs', a traceback might be occurs saying the other_costs does not exist in a dict used for the sequence to display the project profitability in a certain order.

sentry traceback: - 
```
KeyError: 'other_costs'
  File "odoo/http.py", line 2115, in __call__
    response = request._serve_db()
  File "odoo/http.py", line 1698, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "odoo/service/model.py", line 134, in retrying
    result = func()
  File "odoo/http.py", line 1725, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "odoo/http.py", line 1922, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "odoo/addons/base/models/ir_http.py", line 154, in _dispatch
    result = endpoint(**request.params)
  File "odoo/http.py", line 715, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "addons/web/controllers/dataset.py", line 28, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "addons/web/controllers/dataset.py", line 24, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "odoo/api.py", line 461, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "odoo/api.py", line 448, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "addons/sale_timesheet/models/project.py", line 301, in get_panel_data
    panel_data = super(Project, self).get_panel_data()
  File "addons/sale_project/models/project.py", line 273, in get_panel_data
    panel_data = super().get_panel_data()
  File "home/odoo/src/enterprise/saas-16.2/project_account_budget/models/project.py", line 61, in get_panel_data
    panel_data = super().get_panel_data()
  File "addons/project/models/project.py", line 881, in get_panel_data
    profitability_items = self._get_profitability_items()
  File "addons/sale_timesheet/models/project.py", line 494, in _get_profitability_items
    return self._get_profitability_items_from_aal(
  File "addons/sale_timesheet/models/project.py", line 480, in _get_profitability_items_from_aal
    {'data': convert_dict_into_profitability_data(costs_dict), 'total': total_costs},
  File "addons/sale_timesheet/models/project.py", line 441, in convert_dict_into_profitability_data
    data = {'id': invoice_type, 'sequence': sequence_per_invoice_type[invoice_type], **vals}
```
Applying these changes will resolve this issue.

sentry: - 4204230673


---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/122609).

sabr-odoo commented:
Hello, @odoo/sentry_reviewers  Can you please look into this PR?

Thank You!

auon-odoo reviewed (changes requested):
Hello!
Can you please rework your commit message to have 4 distinct sections:
- Steps to reproduce
- Issue
- Cause of the issue (technical, in detail)
- Fix (and explain why it is the best solution)

I was not able to reproduce, and didn't understand what was expected

auon-odoo commented:
@damrOdoo :eyes:

abz-odoo commented:
@sabr-odoo

xavierbol reviewed (approved):
LGTM, thanks for your work. 🙂 

robodoo r+

robodoo commented:
I'm sorry, @xavierbol: draft PRs can not be approved.

xavierbol commented:
Hello @sabr-odoo ,
@auon-odoo is right can you reword the commit message to correctly understand how to reproduce the issue? 

Thanks in advance! 🙂

sabr-odoo commented:
Hello @auon-odoo, 
fixed this issue base on given data from the sentry platform and that data is provided below : 

d | 
{billable_fixed:| {billed: 0.0, record_ids: [8], to_bill: 0.0}, other_costs: {billed: -300.0, to_bill: 0.0}}
-- | --
get_timesheets_action | <function Project._get_profitability_items_from_aal.<locals>.get_timesheets_action at 0x7f8ace152ef0>
invoice_type | 'other_costs'
key1 | 'to_bill'
key2 | 'billed'
profitability_data | []
record_ids | []
vals | {billed: -300.0, to_bill: 0.0}

after that, we fixed this issue by adding squence_id of `other_cost` due to  this error came from `data = {'id': invoice_type, 'sequence': sequence_per_invoice_type[invoice_type], **vals}` this LOC , so `sequence_per_invoice_type` is calling from this `_get_profitability_sequence_per_invoice_type` function and in that time  `other_cost`  squence_id is missing.

xavierbol commented:
Thanks for the explanation 

robodoo r+

xavierbol commented:
In fact, it was more an explanation to explain how to reproduce the issue in the interface

mart-e commented:
Hello,

Please don't modify .pot files by hand, use the export translation wizard.
This .pot is invalid (duplicated entry) and broke the synchronisation.
Fixing at #125182 
cc @oomsveta
